src/controllers/items.py

```diff
@@ -230,6 +232,77 @@ async def add_items(request: Request, imdb_ids: str = None) -> MessageResponse:
     return {"message": f"Added {len(valid_ids)} item(s) to the queue"}
 
 
+@router.post(
+    "/add-manually",
+    summary="Add Media Items Manually",
+    description="Add media item manually with a magnet link or infohash",
+    operation_id="add_item_manually",
+)
+async def add_item_manually(request: Request, imdb_id: str = None, input: str = None) -> MessageResponse:
+    if not imdb_id:
+        raise HTTPException(status_code=400, detail="No IMDb ID provided")
+
+    if not imdb_id.startswith("tt"):
+        raise HTTPException(status_code=400, detail="No valid IMDb ID provided")
+    
+    type, infohash = get_type_and_infohash(input)
+
+    if not infohash:
+        raise HTTPException(status_code=400, detail="No valid input provided")
+
+    downloader: Downloader = request.app.program.services.get(Downloader)
+    with db.Session() as session: 
+        item = create_item_from_imdb_id(imdb_id)
+        item.requested_by = "user"
+        item.requested_at = datetime.now()
+
+        needed_media = get_needed_media(item)
+        cached_streams = downloader.get_cached_streams([infohash], needed_media)        
+
+        if len(cached_streams) == 0:
+            session.rollback()
+            raise HTTPException(
+                status_code=400,
+                detail=f"The selected torrent is not cached for {item.log_string}",
+            )
+
+
+        raw_title = f"Manual Torrent for {imdb_id}"
+        torrent = Torrent(
+            raw_title=raw_title,
+            infohash=infohash,
+            data=ParsedData(raw_title=raw_title),
+        )
+        stream = Stream(torrent)
+        session.add(stream)
+        item.streams = [stream]
+        item.active_stream = cached_streams[0]
+        session.add(item)
+        session.commit()
+        
+        try:
+            downloader.download(item, item.active_stream)
+        except Exception as e:
+            logger.error(f"Failed to download {item.log_string}: {e}")
+            if item.active_stream.get("infohash", None):
+                downloader._delete_and_reset_active_stream(item)
+            session.rollback()
+            raise HTTPException(
+                status_code=500, detail=f"Failed to download {item.log_string}: {e}"
+            ) from e
+
+        session.commit()
+
+        request.app.program.em.add_event(Event("Symlinker", item))
```

Comment:
we handle by id's now, so when the id_handling branch is done this will need to change to `item._id`

Reply:
Ah this still needs fixing I believe